pkg/psmdb/statefulset.go

```diff
@@ -417,12 +423,18 @@ func backupAgentContainer(cr *api.PerconaServerMongoDB, replsetName string, port
 					},
 				},
 			},
-			{
-				Name:  "PBM_MONGODB_URI",
-				Value: "mongodb://$(PBM_AGENT_MONGODB_USERNAME):$(PBM_AGENT_MONGODB_PASSWORD)@$(POD_NAME)",
-			},
 		}...)
 
+		mongoDBURI := "mongodb://$(PBM_AGENT_MONGODB_USERNAME):$(PBM_AGENT_MONGODB_PASSWORD)@$(POD_NAME)"
+		if cr.CompareVersion("1.20.0") >= 0 {
+			mongoDBURI = buildMongoDBURI(ctx, tlsEnabled, sslSecret)
+		}
```

Comment:
it's ok for me and we merge like this, but I prefer to use inverted logic: 
```
mongoDBURI = buildMongoDBURI(ctx, tlsEnabled, sslSecret)
if cr.CompareVersion("1.20.0") < 0 {
			mongoDBURI := "mongodb://$(PBM_AGENT_MONGODB_USERNAME):$(PBM_AGENT_MONGODB_PASSWORD)@$(POD_NAME)"
		}
```
It's easier delete unnecessary code when we don't need it.